src/module/nft/bpf/nft.bpf.c

```diff
@@ -49,27 +49,52 @@ struct nft_event {
 	u8 policy;
 } __attribute__((packed));
 
+struct nft_rule_dp___5_17_0 {
+	u64	is_last:1,
+		handle:42;
+} __attribute__((preserve_access_index));
+
+struct nft_rule___x {
```

Comment:
> Since we use the kernel version where the struct or member was introduced, we could do the same here. Bonus point is this will be obvious for the reader that this is legacy.
> 

of course, this was the reason for `nft_rule_dp` or when this was first introduced earlier for `nft_rule_dp_last`.
nft_rule was included with:

```96518518cc41 ("netfilter: add nftables")```

which is very old and means smth like 3.13 :)
I was going to prefer a comment stating that this got introduced in an unsupported kernel, but I dropped the idea Yeah, ___x doesn't even tell struct nft_rule is the older one (unless we establish this convention now :).
Either way works. ___3_13_0 feels a bit weird (a bit eldelry :), but one can call out consistency.

Reply:
> > which is very old and means smth like 3.13 :) I was going to prefer a comment stating that this got introduced in an unsupported kernel, but I dropped the idea. Yeah, I agree that ___x fails to tellthat struct nft_rule is the older one (unless we establish this convention now :).
> 
> We already use `___x` when we only have one version of the struct. Maybe we should always mandate a version though.

AFAIC, +1 for this

> 
> > Either way works. ___3_13_0 feels a bit weird (a bit old :), but one can call out consistency.
> 
> That's fine IMHO. It's not about Retis, but about the kernel itself.

agreed